### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [vstan02/conix#4]
Would you like to add more error handling for return values from functions like the following?
* [malloc](https://pubs.opengroup.org/onlinepubs/9699919799/functions/malloc.html "Memory allocation") ⇒ [cnx_cli_init](https://github.com/vstan02/conix/blob/0c3e378ad664d15cd7d154d5d23816e39c1cbc99/src/conix.c#L35)
* [strdup](https://pubs.opengroup.org/onlinepubs/9699919799/functions/strdup.html "Duplicate a string") ⇒ [push](https://github.com/vstan02/conix/blob/3a3fa8242c2907f0bd587756f0f90c714d9bfc0b/src/info.c#L65)

**Comment by vstan02** [comment]
@elfring Yep, my bad. Thank you for adding this issue.

**Comment by elfring** [comment]
:thinking: How do you think about to improve [static source code analysis](https://en.wikipedia.org/wiki/List_of_tools_for_static_code_analysis#C.2C_C.2B.2B "Tools for code analysis") also for this software?

**Comment by vstan02** [comment]
@elfring Good idea. If you have any ideas about this, please create a new issue, where we can discuss them (and after implementing them).

**Comment by elfring** [comment]
:thought_balloon: Would you become interested in [aspect-oriented software development](https://en.wikipedia.org/wiki/Aspect-oriented_programming "Documentation")?
